src/main/kotlin/com/sourcegraph/cody/chat/NewChatAction.kt

```diff
@@ -17,6 +18,12 @@ class NewChatAction : DumbAwareAction() {
     showToolbar(project)
   }
 
+  override fun update(event: AnActionEvent) {
+    val project = event.project ?: return
+    val hasActiveAccount = !CodyAuthenticationManager.instance.hasNoActiveAccount(project)
```

Comment:
How often is update called? And how expensive `CodyAuthenticationManager.instance.hasNoActiveAccount` is?
Is that called on EDT?

Reply:
Update is called very often: Any interaction with a toolbar fires this method few times per second (clicking, hovering with mouse, etc).

It's fast and "offline" service with persistent state. It's `cody_accounts.xml`.